compiler.h changes for Windows MSVC support

tarun292 opened the pull request:
Differential Revision: D65328573

malfet commented on kernels/prim_ops/register_prim_ops.cpp:
```diff
@@ -148,7 +148,7 @@ static Kernel prim_ops[] = {
           if (a.isInt() && b.isInt()) {
             const int64_t quot = a.toInt() / b.toInt();
-            if (std::signbit(a.toInt()) == std::signbit(b.toInt())) {
+            if ((a.toInt() < 0) == (b.toInt() < 0)) {
```
This feels unrelated to PR in question, is it?

tarun292 committed:
compiler.h changes for Windows MSVC support (#6623)

Summary:
Adding some MSVC (Windows) specific macros to compiler.h to make sure we can build with MSVC.



Reviewed By: dbort

Differential Revision: D65328573

chinazhangchao commented on runtime/platform/compiler.h:
```diff
@@ -100,22 +100,38 @@
+// There currently doesn't seem to be a great way to do this in Windows and
+// given that weak linkage is not really critical on Windows, we'll just leave
+// it as a stub.
+#define ET_WEAK
```
Hi, we are adding `Windows x64 Build Support` https://github.com/pytorch/executorch/pull/6979. I use `.\install_requirements.bat --pybind xnnpack` to build on windows. And we encountered duplicate symbol problem. 
![image](https://github.com/user-attachments/assets/f3a3ca3d-b3ad-4d3a-8691-5abf120404dc)
I investigated and found that windows didn't support weak symbol. Do you have some insights how to resolve this?
